src/utilities/ToastQueueManager.ts

```diff
@@ -0,0 +1,66 @@
+import { AxiosError } from "axios";
+import Toast, { ToastOptions } from "react-native-root-toast";
+import { ServerErrorResponse } from "../TypesAndInterfaces/config-sync/api-type-sync/toast-types";
+
+const DefaultToastConfig:ToastOptions = {
+    duration: Toast.durations.LONG
+}
+
+interface ToastQueueManagerParams {
+    message: string,
+    options?: ToastOptions
+}
+
+class ToastQueueManager {
+
+    queue: ToastQueueManagerParams[];
+    isToastVisible: boolean
+
+    constructor() {
+        this.queue = [];
+        this.isToastVisible = false;
+    }
+    show({error, options, message}: {error?: AxiosError<ServerErrorResponse>; options?:ToastOptions; message?:string}): void {
```

Comment:
This is good, but you'll have to go through all the usages and pass in named arguments.
I think the compiler might be confused since they're both objects.  
But this should not work with the new format:
```
}).catch((error:AxiosError<ServerErrorResponse>) => ToastQueueManager.show(error));
```
Should be: 
`ToastQueueManager.show({error})` since the names match otherwise:  `ToastQueueManager.show({error: axiosError})`